src/database/schema/ChannelListSchema.ts

```diff
@@ -369,6 +486,25 @@ class ChannelList implements BaseDbSchema {
       members: null
     });
   }
+
+  static fromMessageSignedAPI(data: MessageAnonymouslyData): ChannelList {
+    const isGroup = data?.members?.length > 2;
+    return new ChannelList({
+      id: data?.channel?.id,
+      channelPicture: isGroup ? null : data?.appAdditionalData?.targetImage,
+      name: data?.appAdditionalData?.targetName,
+      description: data?.appAdditionalData?.message,
+      unreadCount: 0,
+      channelType: isGroup ? CHANNEL_GROUP : PM,
+      lastUpdatedAt: data?.channel?.last_message_at,
+      lastUpdatedBy: '',
+      createdAt: data?.channel?.created_at,
+      rawJson: data?.appAdditionalData?.rawJson,
+      user: null,
+      expiredAt: null,
+      members: null
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `fromMessageSignedAPI` function seems to be implemented correctly. It creates a new `ChannelList` object from the provided data and sets the `channelType` based on the number of members in the channel.


<!-- This is an auto-generated comment by OSS CodeRabbit -->